cocos/core/data/class.ts

```diff
@@ -455,7 +455,7 @@ function parseAttributes (constructor: Function, attributes: PropertyStash, clas
         else if (typeof type === 'object') {
             if (Enum.isEnum(type)) {
                 (attrs || initAttrs())[`${propertyNamePrefix}type`] = ENUM_TAG;
-                attrs![`${propertyNamePrefix}enumList`] = Enum.getList(type);
+                attrs![`${propertyNamePrefix}enumType`] = type;
```

Comment:
Currently, many developers, especially plugin authors, directly use the `enumList` variable to control the dropdown list in the inspector. How can we ensure compatibility after making changes?

Reply:
How about add a new record here instead of replace it?

Reply:
Should have been fixed here: https://github.com/cocos/cocos-engine/pull/15029/commits/59666a14b5c643b105827f1022d24078e3654731